app/models/manageiq/providers/nuage/network_manager.rb

```diff
@@ -91,4 +91,53 @@ def l3_cloud_subnets
   def l2_cloud_subnets
     cloud_subnets.where(:type => self.class.l2_cloud_subnet_type)
   end
+
+  def ansible_env_vars
+    {}
+  end
+
+  def ansible_extra_vars(extra)
+    {
+      :nuage_auth => {
+        :api_username   => default_authentication.userid,
+        :api_password   => default_authentication.password,
```

Comment:
does the module take these as ENV vars? I think the extravars can be logged, so it's better to pass secrets through env vanrs

Reply:
@Ladas module doesn't recognize ENV vars but fortunately it setts `no_log=True` for sensitive parameters which prevents printing it to the log: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nuage/nuage_vspk.py#L977.

BTW I'm opening a new PR today with only DELETE methods so that we don't rely on UI at this point.